libs/sdk-ui-kit/src/Form/InputPure.tsx

```diff
@@ -0,0 +1,205 @@
+// (C) 2020 GoodData Corporation
+import React from "react";
+import cx from "classnames";
+import NativeListener from "react-native-listener";
+import bindAll from "lodash/bindAll";
+import noop from "lodash/noop";
+
+import { ENUM_KEY_CODE } from "./typings";
+
+/**
+ * @internal
+ */
+
+export interface InputPureProps {
+    autofocus: boolean;
+    className: string;
+    clearOnEsc: boolean;
+    disabled: boolean;
+    hasError: boolean;
+    hasWarning: boolean;
+    isSearch: boolean;
+    isSmall: boolean;
+    maxlength: number;
+    onChange: (value: string | number, e?: React.ChangeEvent<HTMLInputElement>) => void;
+    onEscKeyPress: () => void;
+    onEnterKeyPress: () => void;
+    onBlur: () => void;
+    onFocus: () => void;
+    placeholder: string;
+    prefix: string;
+    readonly: boolean;
+    suffix: string;
+    label: string;
+    labelPositionTop: boolean;
+    value: string | number;
+}
+/**
+ * @internal
+ */
+export class InputPure extends React.PureComponent<InputPureProps> {
+    public inputNodeRef: HTMLInputElement;
```

Comment:
Can we please use `createRef`? The callback refs were deprecated and this would be a good opportunity to make this change (it was too risky for goodstrap).

Reply:
these refs are leaking and used outside the component